org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ServerLauncher.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2016, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ide.server;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.util.concurrent.Future;
+
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.InputOutput;
+
+import com.google.common.base.Objects;
+import com.google.common.io.ByteStreams;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * @since 2.11
+ */
+public class ServerLauncher {
+	public static final String LOG = "-log";
+
+	public static final String TRACE = "-trace";
+
+	public static final String NO_VALIDATE = "-noValidate";
+
+	public static void main(String[] args) {
+		ServerLauncher.launch(ServerLauncher.class.getName(), args, new ServerModule());
+	}
+
+	public static void launch(String prefix, String[] args, com.google.inject.Module... modules) {
+		LaunchArgs launchArgs = ServerLauncher.createLaunchArgs(prefix, args);
+		ServerLauncher launcher = Guice.createInjector(modules).<ServerLauncher>getInstance(ServerLauncher.class);
+		launcher.start(launchArgs);
+	}
+
+	@Inject
+	private LanguageServerImpl languageServer;
+
+	public void start(LaunchArgs it) {
+		try {
+			InputOutput.println("Xtext Language Server is starting.");
+			Launcher<LanguageClient> launcher = Launcher.createLauncher(languageServer,
+					LanguageClient.class, it.getIn(), it.getOut(), it.isValidate(), it.getTrace());
+			languageServer.connect(launcher.getRemoteProxy());
+			Future<Void> future = launcher.startListening();
+			InputOutput.println("Xtext Language Server has been started.");
+			while (!future.isDone()) {
+				Thread.sleep(10_000l);
+			}
+		} catch (InterruptedException e) {
+			throw Exceptions.sneakyThrow(e);
```

Comment:
The SocketServerLauncher prints the stacktrace instead.

Reply:
yes but that one does not stream redirect and craps
=> want to keep as was for now